### PR TITLE
Add Windows ARM64 support

### PR DESCRIPTION
**Pull request by dennisameling**

### CODE REVIEW

File: example/console-example/WinToast Console Example.vcxproj [hunk 1]
```diff
@@ -17,18 +17,26 @@
...
+      <Configuration>Release</Configuration>
+      <Platform>ARM64</Platform>
+    </ProjectConfiguration>
   </ItemGroup>
   <PropertyGroup Label="Globals">
     <VCProjectVersion>15.0</VCProjectVersion>
     <ProjectGuid>{7C5AC60D-8668-47B6-9D05-FB363F854065}</ProjectGuid>
     <Keyword>Win32Proj</Keyword>
-    <WindowsTargetPlatformVersion>8.1</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
```
Comment (dscho) [inline example/console-example/WinToast Console Example.vcxproj:33]: Will this change anything with regard to Windows 8 support?

If so, would it make sense to override that platform in a `<PropertyGroup Include="*|ARM64">`-like section instead?

File: example/console-example/WinToast Console Example.vcxproj [hunk 2]
```diff
@@ -17,18 +17,26 @@
...
   </PropertyGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
   <PropertyGroup Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v140</PlatformToolset>
+    <PlatformToolset>v143</PlatformToolset>
```
Comment (dscho) [inline example/console-example/WinToast Console Example.vcxproj:39]: Same comment here: would this break Windows 8 support?

File: example/console-example/WinToast Console Example.vcxproj (hunk 1 above)
Comment (dennisameling) [reply example/console-example/WinToast Console Example.vcxproj:33]: It shouldn't break Windows 8 support. The [upstream repo](https://github.com/mohabouje/WinToast/commit/017cf350c1350a3ff2afabe95bfdf89d9e73da05) also uses this TargetPlatform already. I'll build on a Windows 11 machine and will then test the resulting binary on a Windows 8.1 VM (currently downloading an ISO for it).

Comment (Alovchin91) [reply example/console-example/WinToast Console Example.vcxproj:33]: `WindowsTargetPlatformVersion`, despite its name, specifies the version of the SDK to use. 10.x SDK does support Windows 8.1: https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/

File: example/console-example/WinToast Console Example.vcxproj (hunk 2 above)
Comment (Alovchin91) [reply example/console-example/WinToast Console Example.vcxproj:39]: This only specifies the version of the toolset, e.g. the compiler. The toolset still supports Windows 8.1: https://learn.microsoft.com/en-us/cpp/overview/supported-platforms-visual-cpp?view=msvc-170

File: example/console-example/WinToast Console Example.vcxproj (hunk 1 above)
Comment (dennisameling) [reply example/console-example/WinToast Console Example.vcxproj:33]: @dscho I just tested this on Windows 8.1. The version I created using the Windows 10.x SDK and PlatformToolset=v143 is working "correctly". That is, it has the exact same behavior as the version that comes with GfW today.

Current version (unmodified by me) that pops up when installing GfW 2.38.0 and running the `Git for Windows Updater` scheduled task:

image

The version I created using the toolset mentioned above (binaries can be found [here](https://github.com/dennisameling/WinToast/releases/tag/test-dennis)):

```
$ "./WinToast.Console.Example.x64.exe" --appname "Git for Windows" \
--appid GitForWindows.Updater \
--image /mingw64/share/git/git-for-windows.ico \
--text "Download and install Git for Windows DEMO?" \
--action Yes --action No --expirems 15000
```

image

So for the scope of this PR, everything works correctly. However, it looks like WinToast can't deal with .ico files on Windows 8.1. When I invoke the toast with a quick and dirty `.png` file, the logo is shown correctly: 

image

Is that something you want to have fixed at some point? I could provide a follow-up PR in [build-extra](https://github.com/git-for-windows/build-extra/blob/afbd82e9475f1f9f81760f6100e2d71d829c04c5/git-extra/git-update-git-for-windows#L297-L301) to fix the logo thing, but would need a proper `.png` version of the logo first.

Thanks!

Comment (rimrul) [reply example/console-example/WinToast Console Example.vcxproj:33]: Seeing as the unmodified binary produces the same result, I'd say the Icon is probably unrelated to this PR.

Comment (dennisameling) [reply example/console-example/WinToast Console Example.vcxproj:33]: Unrelated indeed. Just wanted to mention it here to make you aware 👍🏼

Comment (dscho) [reply example/console-example/WinToast Console Example.vcxproj:33]: @dennisameling thank you for digging deeper. I agree that it is unrelated, and if you find the time to open that follow-up PR in build-extra, I will be delighted to merge it!

**Pull request merged**